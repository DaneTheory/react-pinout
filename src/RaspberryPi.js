import React, { Component } from 'react';

import Picto from './Picto';
import defaultPinout from './defaultPinout.json'

import './style.css';

const Pin = ({ id, className, href, title, label, details, physicalPort, isActive, shape, color, pictoStyle={} }) => (
  <li className={ className }>
    <a href={ href } title={ title }>
      <span className="default">
        <span className="phys">{ physicalPort }</span>
        { label }
        { details && <small> { details }</small> }
      </span>
      <Picto shape={ shape } color={ color } style={ pictoStyle } />
    </a>
  </li>
)

const RaspberryPi = ({ pins }) => (
  <div className="Pinout">
    <div className="gpio">
      <div className="header"></div>
      <ul className="left">
        { pins.filter((p, i) => (i % 2 === 0)).map((pin, i) => <Pin key={ i } { ...pin }/>) }
      </ul>
      <ul className="right">
        { pins.filter((p, i) => (i % 2 === 1)).map((pin, i) => <Pin key={ i } { ...pin }  pictoStyle={ { left: 2, top: 2 } } />) }
      </ul>
    </div>
  </div>
)
RaspberryPi.defaultProps = {
  pins: defaultPinout
}


export default RaspberryPi;
