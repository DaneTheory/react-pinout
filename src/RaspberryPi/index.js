// @flow
import React from 'react';

import Pin from './Pin';
import defaultPinout from './defaultPinout.json'
import {
  Pinout,
  Title,
  PinsHeader,
  PinsLeft,
  PinsRight
} from './styled'

const RaspberryPi = ({ pins, title, onClick } : { pins: Array<Object>, title: String, onClick: Function }) => {
  const leftColPins = pins.filter((p, i) => (i % 2 === 0));
  const rightColPins = pins.filter((p, i) => (i % 2 === 1));
  return (
    <Pinout>
      <Title>{ title }</Title>
      <PinsHeader />
      <PinsLeft>
        { leftColPins.map((pin, i) => <Pin onClick={ () => onClick(pin) } key={ i } { ...pin }/>) }
      </PinsLeft>
      <PinsRight>
        { rightColPins.map((pin, i) => <Pin onClick={ () => onClick(pin) } key={ i } { ...pin } pictoStyle={ { left: 2, top: 2 } } />) }
      </PinsRight>
    </Pinout>
  )
}
RaspberryPi.propTypes = {
  pins: React.PropTypes.array.isRequired,
  title: React.PropTypes.string,
  onClick: React.PropTypes.func
}
RaspberryPi.defaultProps = {
  pins: defaultPinout,
  title: 'Raspberry Pi Pinout',
  onClick: () => {}
}


export default RaspberryPi;
