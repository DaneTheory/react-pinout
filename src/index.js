
import RaspberryPi from './RaspberryPi';

export RaspberryPi from './RaspberryPi';

export default RaspberryPi;