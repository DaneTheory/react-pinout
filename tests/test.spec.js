import React from 'react'
import { RaspberryPi } from '../src'

import renderer from 'react-test-renderer';

test('RaspberryPi renders correctly', () => {
  const tree = renderer.create(
    <RaspberryPi />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});