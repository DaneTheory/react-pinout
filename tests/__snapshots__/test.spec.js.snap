exports[`test RaspberryPi renders correctly 1`] = `
<div
  className="bksEyz">
  <div
    className="gzEDUL">
    Raspberry Pi Pinout
  </div>
  <div
    className="gbJbiW" />
  <div
    className="hrUUfK">
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          1
        </span>
        3v3 Power
        
      </span>
      <span
        style={
          Object {
            "background": "#B58900",
            "border": "1px solid transparent",
            "borderRadius": 0,
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 8">
      <span>
        <span
          className="phys">
          3
        </span>
        BCM 2
        <small>
           
          (SDA)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#268BD2",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 9">
      <span>
        <span
          className="phys">
          5
        </span>
        BCM 3
        <small>
           
          (SCL)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#268BD2",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 7">
      <span>
        <span
          className="phys">
          7
        </span>
        BCM 4
        <small>
           
          (GPCLK0)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          9
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 0">
      <span>
        <span
          className="phys">
          11
        </span>
        BCM 17
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 2">
      <span>
        <span
          className="phys">
          13
        </span>
        BCM 27
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 3">
      <span>
        <span
          className="phys">
          15
        </span>
        BCM 22
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          17
        </span>
        3v3 Power
        
      </span>
      <span
        style={
          Object {
            "background": "#B58900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 12">
      <span>
        <span
          className="phys">
          19
        </span>
        BCM 10
        <small>
           
          (MOSI)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 13">
      <span>
        <span
          className="phys">
          21
        </span>
        BCM 9
        <small>
           
          (MISO)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 14">
      <span>
        <span
          className="phys">
          23
        </span>
        BCM 11
        <small>
           
          (SCLK)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          25
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 30">
      <span>
        <span
          className="phys">
          27
        </span>
        BCM 0
        <small>
           
          (ID_SD)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#268BD2",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 21">
      <span>
        <span
          className="phys">
          29
        </span>
        BCM 5
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 22">
      <span>
        <span
          className="phys">
          31
        </span>
        BCM 6
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 23">
      <span>
        <span
          className="phys">
          33
        </span>
        BCM 13
        <small>
           
          (PWM1)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 24">
      <span>
        <span
          className="phys">
          35
        </span>
        BCM 19
        <small>
           
          (MISO)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 25">
      <span>
        <span
          className="phys">
          37
        </span>
        BCM 26
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          39
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
  </div>
  <div
    className="jUDAZX">
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          2
        </span>
        5v Power
        
      </span>
      <span
        style={
          Object {
            "background": "#DC322F",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          4
        </span>
        5v Power
        
      </span>
      <span
        style={
          Object {
            "background": "#DC322F",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          6
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 15">
      <span>
        <span
          className="phys">
          8
        </span>
        BCM 14
        <small>
           
          (TXD)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#6c71c4",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 16">
      <span>
        <span
          className="phys">
          10
        </span>
        BCM 15
        <small>
           
          (RXD)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#6c71c4",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 1">
      <span>
        <span
          className="phys">
          12
        </span>
        BCM 18
        <small>
           
          (PWM0)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          14
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 4">
      <span>
        <span
          className="phys">
          16
        </span>
        BCM 23
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 5">
      <span>
        <span
          className="phys">
          18
        </span>
        BCM 24
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          20
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 6">
      <span>
        <span
          className="phys">
          22
        </span>
        BCM 25
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 10">
      <span>
        <span
          className="phys">
          24
        </span>
        BCM 8
        <small>
           
          (CE0)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 11">
      <span>
        <span
          className="phys">
          26
        </span>
        BCM 7
        <small>
           
          (CE1)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 31">
      <span>
        <span
          className="phys">
          28
        </span>
        BCM 1
        <small>
           
          (ID_SC)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#268BD2",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          30
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 26">
      <span>
        <span
          className="phys">
          32
        </span>
        BCM 12
        <small>
           
          (PWM0)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="">
      <span>
        <span
          className="phys">
          34
        </span>
        Ground
        
      </span>
      <span
        style={
          Object {
            "background": "#002B36",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 27">
      <span>
        <span
          className="phys">
          36
        </span>
        BCM 16
        
      </span>
      <span
        style={
          Object {
            "background": "#859900",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 28">
      <span>
        <span
          className="phys">
          38
        </span>
        BCM 20
        <small>
           
          (MOSI)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
    <div
      onClick={[Function]}
      title="Wiring Pi pin 29">
      <span>
        <span
          className="phys">
          40
        </span>
        BCM 21
        <small>
           
          (SCLK)
        </small>
      </span>
      <span
        style={
          Object {
            "background": "#D33682",
            "border": "1px solid transparent",
            "borderRadius": "50%",
            "display": "block",
            "height": 16,
            "left": 2,
            "position": "absolute",
            "right": 2,
            "top": 2,
            "width": 16,
          }
        }>
        <span
          style={
            Object {
              "background": "#FDF6E3",
              "borderRadius": "100%",
              "display": "block",
              "fontSize": 7,
              "height": 6,
              "left": 5,
              "position": "absolute",
              "top": 5,
              "width": 6,
            }
          } />
      </span>
    </div>
  </div>
</div>
`;
